iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -16,38 +16,97 @@
 
 #include "iceoryx_hoofs/posix_wrapper/thread.hpp"
 #include "iceoryx_hoofs/cxx/helplets.hpp"
+#include "iceoryx_hoofs/internal/log/hoofs_logging.hpp"
 #include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
 
 namespace iox
 {
 namespace posix
 {
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept
+Thread::~Thread() noexcept
 {
-    posixCall(iox_pthread_setname_np)(thread, name.c_str()).successReturnValue(0).evaluate().or_else([](auto& r) {
-        // String length limit is ensured through cxx::string
-        // ERANGE (string too long) intentionally not handled to avoid untestable and dead code
-        std::cerr << "This should never happen! " << r.getHumanReadableErrnum() << std::endl;
-        cxx::Ensures(false && "internal logic error");
-    });
+    if (m_isJoinable)
+    {
+        auto joinResult = posixCall(pthread_join)(m_threadHandle, nullptr).successReturnValue(0).evaluate();
+        if (joinResult.has_error())
+        {
+            switch (joinResult.get_error().errnum)
+            {
+            case EDEADLK:
+                LogError() << "A deadlock was detected when attempting to join the thread.";
+                break;
+            default:
+                LogError() << "This should never happen. An unknown error occurred.";
+                break;
+            }
+        }
+        m_isJoinable = false;
+    }
 }
 
-ThreadName_t getThreadName(pthread_t thread) noexcept
+void Thread::setThreadName(const ThreadName_t& name) noexcept
+{
+    posixCall(iox_pthread_setname_np)(m_threadHandle, name.c_str())
+        .successReturnValue(0)
+        .evaluate()
+        .or_else([](auto& r) {
+            // String length limit is ensured through cxx::string
+            // ERANGE (string too long) intentionally not handled to avoid untestable and dead code
+            LogError() << "This should never happen! " << r.getHumanReadableErrnum();
+            cxx::Ensures(false && "internal logic error");
```

Comment:
Yes, maybe it is too harsh but the error handling here has to be reworked anyway since the possible `open` errors are not considered. Therefore, I'd like to postpone the decision on how to react when setting the thread name fails until this is implemented.